syz-verifier/main.go

```diff
@@ -405,9 +405,9 @@ func (srv *RPCServer) NextExchange(a *rpctype.NextExchangeArgs, r *rpctype.NextE
 // newResult is called when a Runner sends a new Result. It returns true if all
 // Results from the corresponding programs have been received and they can be
 // sent for verification. Otherwise, it returns false.
-func (srv *RPCServer) newResult(res *Result, prog *progInfo) bool {
-	prog.res = append(prog.res, res)
-	delete(prog.left, res.Pool)
+func (srv *RPCServer) newResult(pool int, res *Result, prog *progInfo) bool {
+	prog.res[pool] = res
```

Comment:
res already contains pool, so passing pool separate looks excessive. This can be `prog.res[res.Pool] = res`.